pip/req/req_install.py

```diff
@@ -1201,21 +1165,10 @@ def parse_editable(editable_req, default_vcs=None):
             ' is currently supported'
         raise InstallationError(error_message)
 
-    try:
-        options = _build_editable_options(editable_req)
-    except Exception as exc:
+    package_name = Link(url).egg_fragment
+    if not package_name:
         raise InstallationError(
-            '--editable=%s error in editable options:%s' % (editable_req, exc)
+            '--editable=%s is not the right format; it must have '
+            '#egg=Package' % editable_req
         )
-    if not options or 'egg' not in options:
-        req = _build_req_from_url(editable_req)
```

Comment:
Hmm it looks like this should not go away (or it should be indicated as backward incompatible).